ui/src/app/shared/utils/url-search-params.utils.ts

```diff
@@ -64,18 +65,31 @@ import {TimeFrame} from "../model/TimeFrame";
 
   /** Accepts a string query that was generated via encodeUrlSearchParams and
    *  converts it back into a QueryJobsRequest. */
-  public static unpackURLSearchParams(query: string): QueryJobsRequest {
+  public static unpackURLSearchParams(query: string, capabilities?: CapabilitiesResponse): QueryJobsRequest {
     let urlSearchParams = new URLSearchParams(query);
     let queryRequest = {
       labels: {},
       extensions: {}
     };
 
     urlSearchParams.paramsMap.forEach((values: string[], key: string) => {
-      if (queryDataTypes.has(key) || queryExtensionsDataTypes.has(key)) {
-        // If this is a known field, handle the data type explicitly
-        var value: any;
-        let dataType = queryDataTypes.has(key) ? queryDataTypes.get(key) : queryExtensionsDataTypes.get(key);
+      let labelField: DisplayField = null;
+      if (capabilities) {
+        labelField = capabilities.displayFields.find(f => f.field === 'labels.' + key);
+      }
+      if (queryDataTypes.has(key) || queryExtensionsDataTypes.has(key) || labelField) {
+        // If this is label, use the settings to determine type
```

Comment:
I'm not necessarily opposed, but why do we need to do this? The fallback is just to treat any unknown as a label